### PR TITLE
Jenkins 25249

### PR DESCRIPTION
**Pull request by Dohbedoh**
@reviewbybees Changes to make the P4 Trigger pipeline compatible. Added some tests of the P4 trigger for both Pipeline/Non Pipeline jobs.

### CODE REVIEW

**Comment by p4paul** [comment]
Thanks for the changes and especially the tests.

File: src/main/java/org/jenkinsci/plugins/p4/trigger/P4Trigger.java [hunk 1]
```diff
@@ -110,15 +109,15 @@ public File getLogFile() {
...
-		if (job instanceof AbstractProject) {
-			AbstractProject<?, ?> project = (AbstractProject<?, ?>) job;
-			SCM scm = project.getScm();
-
-			if (scm instanceof PerforceScm) {
-				PerforceScm p4scm = (PerforceScm) scm;
+		//Get all the trigger for this Job
+		SCMTriggerItem item = SCMTriggerItem.SCMTriggerItems.asSCMTriggerItem(job);
+		//As soon as we find a match, return
+		for (SCM scmTrigger : item.getSCMs()) {
```
Comment (jglick) [inline src/main/java/org/jenkinsci/plugins/p4/trigger/P4Trigger.java]: :bug: you ignored the `@CheckForNull` on the return value!

**Comment by jglick** [comment]
And @p4paul you ignored what the PR builder was trying to warn you about. :-)

```
[INFO] BugInstance size is 2
[INFO] Error size is 0
[INFO] Total bugs: 2
[INFO] Possible null pointer dereference in org.jenkinsci.plugins.p4.trigger.P4Trigger.matchServer(Job, String) due to return value of called method [org.jenkinsci.plugins.p4.trigger.P4Trigger, org.jenkinsci.plugins.p4.trigger.P4Trigger] Dereferenced at P4Trigger.java:[line 120]Known null at P4Trigger.java:[line 120] NP_NULL_ON_SOME_PATH_FROM_RETURN_VALUE
[INFO] Possible null pointer dereference in org.jenkinsci.plugins.p4.trigger.P4Trigger.matchServer(Job, String) due to return value of called method [org.jenkinsci.plugins.p4.trigger.P4Trigger, org.jenkinsci.plugins.p4.trigger.P4Trigger] Dereferenced at P4Trigger.java:[line 115]Known null at P4Trigger.java:[line 113] NP_NULL_ON_SOME_PATH_FROM_RETURN_VALUE
```

File: src/main/java/org/jenkinsci/plugins/p4/trigger/P4Trigger.java [hunk 2]
```diff
@@ -110,15 +109,15 @@ public File getLogFile() {
...
+			if (scmTrigger instanceof PerforceScm) {
+				PerforceScm p4scm = (PerforceScm) scmTrigger;
 				String id = p4scm.getCredential();
 				P4BaseCredentials credential = ConnectionHelper.findCredential(id);
-				if (port.equals(credential.getP4port())) {
+				if (credential.getP4port().equals(port)) {
```
Comment (jglick) [inline src/main/java/org/jenkinsci/plugins/p4/trigger/P4Trigger.java]: Unclear how this swap is related to the topic of the PR, but FindBugs considers it unsafe.

Comment (jglick) [inline src/main/java/org/jenkinsci/plugins/p4/trigger/P4Trigger.java]: In fact `getP4port` is `@CheckForNull`, so this is definitely a :bug:.

`P4Hook.probeJobs` seems to do no null check on `port`, so it ought to be marked `@CheckForNull` here—or, if this is a mandatory parameter, the web method should be rejected earlier.

Comment (Dohbedoh) [inline src/main/java/org/jenkinsci/plugins/p4/trigger/P4Trigger.java]: Thanks for the review. I opened a new PR to fix this.

**Pull request merged**